purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1048,31 +1048,40 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
             appInBackground,
             { offeringsJSON ->
                 try {
-                    val jsonArrayOfOfferings = offeringsJSON.getJSONArray("offerings")
-                    val skus = mutableSetOf<String>()
-                    for (i in 0 until jsonArrayOfOfferings.length()) {
-                        val jsonPackagesArray =
-                            jsonArrayOfOfferings.getJSONObject(i).getJSONArray("packages")
-                        for (j in 0 until jsonPackagesArray.length()) {
-                            skus.add(
-                                jsonPackagesArray.getJSONObject(j)
-                                    .getString("platform_product_identifier")
-                            )
-                        }
+                    val skus = extractSkus(offeringsJSON)
+                    if (skus.isEmpty()) {
+                        handleErrorFetchingOfferings(
+                            PurchasesError(
+                                PurchasesErrorCode.ConfigurationError,
```

Comment:
this reminds me, there was a request in a `purchases-ios` issue that we update our documentation. it currently says offerings aren't required, but then we error out. we should probably clarify why/when this error is expected. @codykerns thoughts?